src/containers/CreateNote/CreateNote.js

```diff
@@ -14,90 +16,90 @@ export class CreateNote extends React.Component {
       noteItems: this.props.noteItems || [],
       currentFocus: null
     };
-  }
+  };
 
   handleChangeTitle = event => {
     this.setState({ title: event.target.value, currentFocus: null });
   };
 
-  makeCopy = element => JSON.parse(JSON.stringify(element))
-  
   handleChangeNoteItems = event => {
-    const noteItemsCopy = this.makeCopy(this.state.noteItems);
+    const noteItemsCopy = this.makeCopy();
+    const { id } = event.target.closest('label');
     const { value: newText } = event.target;
-    const { id } = event.target.closest('label')
-    const matchedNoteItem = noteItemsCopy.find(note => note.id === id);
-
+    const matchedNoteItem = this.findMatchingNoteItem(noteItemsCopy, id);
     if (matchedNoteItem) {
-      matchedNoteItem.text = newText;     
+      matchedNoteItem.text = newText;
     } else {
       const newListItem = { id, text: newText, isCompleted: false };
       noteItemsCopy.push(newListItem);
     }
+    this.updateNoteItems(noteItemsCopy, id);
+  };
 
-    this.setState({
-      noteItems: noteItemsCopy,
-      currentFocus: id
-    });
+  handleItemDelete = event => {
+    const noteItemsCopy = this.makeCopy();
+    const { id } = event.target.closest('label');
+    const noteItemIndex = noteItemsCopy.findIndex(note => note.id === id);
+    noteItemsCopy.splice(noteItemIndex, 1);
+    this.updateNoteItems(noteItemsCopy, id);
   };
 
   handleToggleIsComplete = event => {
-    const noteItemsCopy = this.makeCopy(this.state.noteItems);
+    const noteItemsCopy = this.makeCopy();
     const { id } = event.target.closest('label');
-    const matchedNoteItem = noteItemsCopy.find(note => note.id === id);
+    const matchedNoteItem = this.findMatchingNoteItem(noteItemsCopy, id);
```

Comment:
Was this change for readability?

Reply:
I think that I originally had planned to have more in this method, but it required too many things being passed around and it would have made it 'reverse refactored'... 
I think it would be more readable (+shorter) if I just got rid of the method again 😜

Reply:
Hahah, I definitely know that feels. Thank you for refactoring!!⭐️